src/sdk/namespace_fs.js

```diff
@@ -877,6 +918,55 @@ class NamespaceFS {
         }
     }
 
+    // Allocated config.NSFS_BUF_SIZE in Semaphore but in fact we can take up more inside
+    // This is due to MD5 calculation and data buffers
+    // Can be finetuned further on if needed and inserting the Semaphore logic inside
+    // Instead of wrapping the whole _upload_stream function (q_buffers lives outside of the data scope of the stream)
+    async _upload_stream_tmp(params, upload_path, file_path, fs_account_config, rpc_client, fs_xattr) {
```

Comment:
@eshelmarc I suggest to merge `_upload_stream_tmp()`  with the existing `_upload_stream()` function, simply by adding another argument which we can name `use_tmpfile` and the callers will send as true/false depending on the path. Inside we can determine the open flags accordingly ('w' vs. 'wt') and also call the linkfileat function before closing (for that you seem to need to send also the file_path to the function which can be done for both call sites).

Reply:
done

Reply:
BTW, you can not close the file before linking it, we use the fd to link the file.